frontend/src/components/activePage/TableWorkspace/TableWorkspace.tsx

```diff
@@ -56,6 +82,19 @@ const TableWorkspace: FC<ITableWorkspaceProps> = ({ ...props }) => {
     );
   };
 
+  const destroySelected = () => {
+    instances
+      .filter(i => selectiveDestroy.includes(i.id))
+      .forEach(instance => {
+        deleteInstanceMutation({
+          variables: {
+            tenantNamespace: instance.tenantNamespace,
+            instanceId: instance.name,
+          },
+        });
+      });
```

Comment:
Would rather show a spinner somewhere (if this operation is from a modal, the confirmation button could be set in a loading state) and make this all async, with a normal for so that it can await the single deletions

Reply:
ok, i need to think the better way to do this, i have some doubts about how to wait the end of all deletions. in the other hand i have for each deletion the loading state

Reply:
The mutation should return a promise, in that case you should be able to await for such promise. You can't await within a forEach though, you need a regular for...

Reply:
In general the issue is that the forEach starts a possibly high number of transactions, with an api call for each of those, which could result in a bad user experience and a rather heavy load on the server side. Handling them one by one instead could be better (I think)...

Reply:
i think the same, if i remember right we already talk about this arg. so i transform it in a For sandard cycle so that i can wait on each cycle the promise (this was my doubt before)